src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -790,87 +790,89 @@ export function useSwapInputsController({
-            isQuoteStale.value = 0;
-            isFetching.value = false;
-            inputValues.modify(values => {
-              return {
-                ...values,
-                inputAmount: 0,
-                inputNativeValue: 0,
-                outputAmount: 0,
-                outputNativeValue: 0,
-              };
-            });
-            return;
-          }
+      if (!didInputAssetChange && !didOutputAssetChange) return;
 
-          if (didInputAssetChange) {
-            inputMethod.value = 'inputAmount';
-            sliderXPosition.value = withSpring(SLIDER_WIDTH / 2, snappySpringConfig);
-          }
+      const balance = internalSelectedInputAsset.value?.maxSwappableAmount;
 
-          const { inputAmount, inputNativeValue } = getInputValuesForSliderPositionWorklet({
-            selectedInputAsset: internalSelectedInputAsset.value,
-            percentageToSwap: didInputAssetChange ? 0.5 : percentageToSwap.value,
-            sliderXPosition: didInputAssetChange ? SLIDER_WIDTH / 2 : sliderXPosition.value,
-          });
+      const areBothAssetsSet = internalSelectedInputAsset.value && internalSelectedOutputAsset.value;
+      const didFlipAssets =
+        didInputAssetChange && didOutputAssetChange && areBothAssetsSet && previous && current.assetToSellId === previous.assetToBuyId;
 
-          inputValues.modify(values => {
-            return {
-              ...values,
-              inputAmount,
-              inputNativeValue,
-            };
-          });
-        } else {
-          // If the assets were flipped
-          inputMethod.value = 'inputAmount';
-
-          const inputNativePrice = internalSelectedInputAsset.value?.nativePrice || internalSelectedInputAsset.value?.price?.value || 0;
-          const outputNativePrice = internalSelectedOutputAsset.value?.nativePrice || internalSelectedOutputAsset.value?.price?.value || 0;
-
-          const prevInputNativeValue = inputValues.value.inputNativeValue;
-          const prevOutputAmount = inputValues.value.outputAmount;
-          const newInputAmount = inputNativePrice > 0 ? divWorklet(prevInputNativeValue, inputNativePrice) : prevOutputAmount;
-
-          const inputAmount = Number(
-            valueBasedDecimalFormatter({
-              amount: newInputAmount,
-              nativePrice: inputNativePrice,
-              roundingMode: 'up',
-              isStablecoin: internalSelectedInputAsset.value?.type === 'stablecoin' ?? false,
-              stripSeparators: true,
-            })
-          );
-
-          const prevOutputNativeValue = inputValues.value.outputNativeValue;
-          const prevInputAmount = inputValues.value.inputAmount;
-          const newOutputAmount = outputNativePrice > 0 ? divWorklet(prevOutputNativeValue, outputNativePrice) : prevInputAmount;
+      if (!didFlipAssets) {
+        // If either asset was changed but the assets were not flipped
+        inputMethod.value = 'inputAmount';
```

Comment:
this line is the main change, pulling it out from where it was previously only updated in the `didInputAssetChange` conditional